src/EPR.Calculator.Frontend/Controllers/CalculationRunNameController.cs

```diff
@@ -14,16 +15,20 @@ public IActionResult Index()
         }
 
         [HttpPost]
-        public IActionResult RunCalculator(string calculationName)
+        public IActionResult RunCalculator(InitiateCalculatorRunModel calculationRunName)
         {
-            if (string.IsNullOrWhiteSpace(calculationName))
+            if (!this.ModelState.IsValid)
             {
-                this.ViewBag.Errors = CreateErrorViewModel();
+                var errorMessages = this.ModelState.Values.SelectMany(x => x.Errors).Select(e => e.ErrorMessage);
+                this.ViewBag.Errors = CreateErrorViewModel(errorMessages.First());
                 return this.View(CalculationRunNameIndexView);
             }
 
-            this.ViewBag.CalculationName = calculationName;
-            this.HttpContext.Session.SetString(SessionConstants.CalculationName, (string)this.ViewBag.CalculationName);
+            this.ViewBag.CalculationName = calculationRunName?.CalculationName;
+            if (!string.IsNullOrWhiteSpace(this.ViewBag.CalculationName))
```

Comment:
`this.ViewBag.CalculationName` can be replaced with `calculationRunName.calculationRunName`
Then you do not need to check for `if (!string.IsNullOrWhiteSpace(this.ViewBag.CalculationName))`

Reply:
Updated